aospy/timedate.py

```diff
@@ -8,113 +8,63 @@
-                       'jfmamjjasond'.  If 'ann', use all months.  Otherwise,
-                       use the specified months.
-        :type months: str or int
-        """
-        assert isinstance(months, (int, str))
-        if isinstance(months, int):
-            return [months]
-        if months.lower() == 'ann':
-            return np.arange(1, 13)
-        first_letter = 'jfmamjjasond' * 2
-        # Python indexing starts at 0; month indices start at 1 for January.
-        count = first_letter.count(months)
-        if (count == 0) or (count > 2):
-            message = ("The user must provide a unique pattern of consecutive "
-                       "first letters of months within '{}'. The provided "
-                       "string '{}' does not comply."
-                       "  For individual months use integers."
-                       "".format(first_letter, months))
-            raise RuntimeError(message)
-        st_ind = first_letter.find(months.lower())
-        return np.arange(st_ind, st_ind + len(months)) % 12 + 1
+def to_datetime(obj, end_of_year=False):
+    """Create datetime object from the inputted object.
+
+    Parameters
+    ----------
+    obj : bool, int, str, or datetime.datetime
+    end_of_year : bool, optional
+        If `obj` is an int, flag indicating whether the desired output
+        corresponds to the start of the year or the end of the year.  Default
+        is False.
+
+    Returns
+    -------
+    datetime.datetime object
+        If `obj` is a bool or datetime.datetime, return `obj` unchanged.  If
+        `obj` is a str, convert to datetime.datetime by first converting to a
+        pandas Timestamp via pd.to_datetime and then to a datetime.datetime via
+        pd.Timestamp.to_datetime.
 
-    def __init__(self, start_date, end_date, months):
-        """Instantiate a TimeManager object."""
-        self.start_date = self.to_datetime(start_date)
-        self.end_date = self.to_datetime(end_date)
-        self.months = self.month_indices(months)
-
-    @staticmethod
-    def _construct_month_conditional(time, months):
-        """Create a conditional statement for selecting data in a DataArray."""
-        cond = False
-        for month in months:
-            cond |= (time['time.month'] == month)
-        return cond
-
-    def create_time_array(self):
-        """Create an xarray.DataArray comprising the desired months."""
-        all_months = pd.date_range(
-            start=self.apply_year_offset(self.start_date),
-            end=self.apply_year_offset(self.end_date), freq='M'
-        )
-        time = xr.DataArray(all_months, dims=[TIME_STR])
-        month_cond = self._construct_month_conditional(time, self.months)
-        return time[month_cond]
+    """
+    if isinstance(obj, (bool, datetime.datetime)):
+        return obj
+    if isinstance(obj, str):
+        return pd.to_datetime(obj).to_datetime()
+    if isinstance(obj, int):
+        if end_of_year:
```

Comment:
In which case there is no need for this function at all, right?  Instead, at the Run and Calc levels directly, we just enforce that the input is of our required datetime type.